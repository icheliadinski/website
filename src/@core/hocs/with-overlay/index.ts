export { withOverlay } from './with-overlay'
