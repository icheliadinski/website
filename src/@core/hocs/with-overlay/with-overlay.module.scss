.overlay {
  background: var(--c-dark);
  height: 100%;
  left: 0;
  opacity: 1;
  position: fixed;
  top: 0;
  width: 100%;
}

.centered {
  align-items: center;
  display: flex;
  justify-content: center;
}
