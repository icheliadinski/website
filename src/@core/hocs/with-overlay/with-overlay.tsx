import React from 'react'
import classNames from 'classnames'
import styles from './with-overlay.module.scss'

export interface Props {
  isCentered?: boolean
}

export const withOverlay = (Component: React.FC) => {
  return ({ isCentered = true, ...props }: Props) => {
    const classes = classNames(styles.overlay, {
      [styles.centered]: isCentered,
    })

    return (
      <div className={classes}>
        <Component {...props} />
      </div>
    )
  }
}
