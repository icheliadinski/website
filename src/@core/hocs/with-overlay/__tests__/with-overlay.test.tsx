import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { withOverlay, Props } from '../with-overlay'

describe('withOverlay', () => {
  const Component = () => <div>Component</div>
  let wrapper: ShallowWrapper<Props>
  let WithOverlay

  beforeEach(() => {
    WithOverlay = withOverlay(Component)
    wrapper = shallow(<WithOverlay />)
  })

  it('renders successfully', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('when not centered', () => {
    beforeEach(() => {
      wrapper.setProps({ isCentered: false })
    })

    it('renders successfully', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
