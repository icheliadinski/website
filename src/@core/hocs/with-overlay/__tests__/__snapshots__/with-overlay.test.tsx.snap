// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withOverlay renders successfully 1`] = `
<div
  className="overlay centered"
>
  <Component />
</div>
`;

exports[`withOverlay when not centered renders successfully 1`] = `
<div
  className="overlay"
>
  <Component />
</div>
`;
